src/tests/compiletest.md

```diff
@@ -4,6 +4,12 @@
 
 ## Introduction
 
+Most of the "meaty" steps that matter are backed by Cargo, which does indeed
+have its own parallelism and incremental management. Later steps, like
+tests, aren't incremental and simply run the entire suite currently.
+However, compiletest itself tries to avoid running tests when the artifacts
+that are involved (mainly the compiler) haven't changed.
```

Comment:
Let's delete everything except "compiletest itself tries to avoid running tests when the artifacts
that are involved (mainly the compiler) haven't changed", and move that section to the end of "Introduction", just before "Test suites".